File: typescript/src/common/domain/DomainEvent.ts
import {v4 as uuidv4} from 'uuid';

export abstract class DomainEvent
{
    public id: string;
    public occurredOn: Date;
    public aggregateId: string;
    public aggregateType: string;

    protected constructor(aggregateId: string, aggregateType: string){
        this.id = uuidv4();
        this.occurredOn = new Date();
        this.aggregateId = aggregateId;
        this.aggregateType = aggregateType;
    }
}

File: kotlin/src/main/kotlin/common/domain/DomainEvent.kt
package common.domain

import java.time.LocalDateTime
import java.util.UUID

abstract class DomainEvent {
    val eventId: String
    val occurredOn: LocalDateTime

    init {
        this.eventId = UUID.randomUUID().toString()
        this.occurredOn = LocalDateTime.now()
    }
}
